packages/web/src/js/TextArea.ts

```diff
@@ -0,0 +1,46 @@
+import BaseComponent from './BaseComponent';
+import { EventHandler, SelectorEngine } from './dom';
+import { debounce, enableToggleAutoloader } from './utils';
+
+const NAME = 'textArea';
+const RESIZE_EVENT = 'resize';
+const DEBOUNCE_DELAY = 50;
+
+class TextArea extends BaseComponent {
+  textarea: HTMLTextAreaElement;
+
+  static get NAME() {
+    return NAME;
+  }
+
+  static get DATA_KEY() {
+    return `${this.NAME}`;
+  }
+
+  constructor(element: HTMLElement) {
+    super(element);
+    this.textarea = SelectorEngine.findOne('textarea', this.element) as HTMLTextAreaElement;
+    this.adjustHeight = debounce(this.adjustHeight, DEBOUNCE_DELAY);
```

Comment:
I think that we should add new property `debouncedAdjustHeight` and not override `this.adjustHeight`.

```suggestion
    this.debouncedAdjustHeight = debounce(this.adjustHeight, DEBOUNCE_DELAY);
```